controllers/sriovoperatorconfig_controller.go

```diff
@@ -409,101 +410,103 @@ func (r *SriovOperatorConfigReconciler) syncK8sResource(cr *sriovnetworkv1.Sriov
 	return nil
 }
 
-func (r *SriovOperatorConfigReconciler) syncOffloadMachineConfig(dc *sriovnetworkv1.SriovOperatorConfig) error {
-	logger := r.Log.WithName("syncOffloadMachineConfig")
+func (r *SriovOperatorConfigReconciler) syncOvsHardwareOffloadMachineConfig(dc *sriovnetworkv1.SriovOperatorConfig) error {
+	logger := r.Log.WithName("syncOvsHardwareOffloadMachineConfig")
 	var err error
 
 	logger.Info("Start to render MachineConfig and MachineConfigPool for OVS HW offloading")
+
 	data := render.MakeRenderData()
-	data.Data["HwOffloadNodeLabel"] = HwOffloadNodeLabel
-	mcName := "00-" + HwOffloadNodeLabel
-	mcpName := HwOffloadNodeLabel
-	mc, err := render.GenerateMachineConfig("bindata/manifests/switchdev-config", mcName, HwOffloadNodeLabel, dc.Spec.EnableOvsOffload, &data)
-	if err != nil {
-		return err
-	}
-	mcpRaw, err := render.RenderTemplate("bindata/manifests/switchdev-config/machineconfigpool.yaml", &data)
+	mcpMap := make(map[string]bool)
+
+	mcpList := &mcfgv1.MachineConfigPoolList{}
+	err = r.List(context.TODO(), mcpList, &client.ListOptions{})
 	if err != nil {
-		return err
-	}
-	mcp := &mcfgv1.MachineConfigPool{}
-	if len(mcpRaw) != 1 {
-		return fmt.Errorf("Invalid MachineConfigPool CR template")
+		return fmt.Errorf("Failed to get MachineConfigPoolList: %v", err)
 	}
-	err = r.Scheme.Convert(mcpRaw[0], mcp, context.TODO())
-	if err != nil {
-		return err
+
+	for _, mcp := range mcpList.Items {
+		selector, err := metav1.LabelSelectorAsSelector(mcp.Spec.NodeSelector)
+		if err != nil {
+			return fmt.Errorf("Invalid label selector in MachineConfigPool: %s, %v", mcp.GetName(), err)
+		}
+		// MachineConfigPool is selected when its NodeSelectors are included in MachineConfigPoolSelector
+		for _, pool := range dc.Spec.OvsHardwareOffload {
+			if selector.Matches(labels.Set(pool.MachineConfigPoolSelector)) {
+				mcpMap[mcp.GetName()] = true
+				break
+			}
+		}
 	}
 
-	foundMC := &mcfgv1.MachineConfig{}
-	foundMCP := &mcfgv1.MachineConfigPool{}
+	// Offload is not supported for master MachineConfigPool
+	delete(mcpMap, "master")
```

Comment:
Instead of deletion, can we check the MCP name before adding to `mcpMap`, and log an error if users select the master pool?

Reply:
Done